components/org.wso2.carbon.identity.application.authenticator.biometric.servlet/src/main/java/org/wso2/carbon/identity/application/authenticator/biometric/servlet/servlet/BiometricServlet.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.wso2.carbon.identity.application.authenticator.biometric.servlet.servlet;
+
+import com.google.gson.Gson;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authenticator.biometric.servlet.javascript.flow.WaitStatusResponse;
+import org.wso2.carbon.identity.application.authenticator.biometric.servlet.model.WaitStatus;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.HashMap;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Component class for implementing the Biometric servlet.
+ */
+public class BiometricServlet extends HttpServlet {
+    private static final Log log = LogFactory.getLog(BiometricServlet.class);
+    private static final String DEVICE_TYPE = "deviceType";
+    private static final String MOBILE = "mobile";
+    private static final String WEB = "web";
+    private static final String SESSION_DATA_KEY_MOBILE = "sessionDataKeyMobile";
+    private static final String SESSION_DATA_KEY_WEB = "sessionDataKeyWeb";
+    private static final String CHALLENGE_MOBILE = "challengeMobile";
+    private HashMap<String, String> updateStatus = new HashMap<>();
+
+    @Override
+    protected void doGet(HttpServletRequest request,
+                         HttpServletResponse response) throws IOException {
+        doPost(request, response);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request,
+                          HttpServletResponse response) throws IOException {
+
+        WaitStatusResponse waitResponse = null;
+        if (request.getParameterMap().containsKey(DEVICE_TYPE)) {
+            String deviceType = request.getParameter(DEVICE_TYPE);
+            waitResponse = new WaitStatusResponse();
+            if (deviceType.equals(MOBILE)) {
+                if ((request.getParameterMap().containsKey(SESSION_DATA_KEY_MOBILE)) &&
```

Comment:
Do we need to have two `SESSION_DATA_KEY_MOBILE` and `SESSION_DATA_KEY_WEB` parameters? Can't we just define a `SESSION_DATA_KEY` parameter and identify WEB and MOBILE scenario using the `DEVICE_TYPE`?